ui/app/AppLayouts/Wallet/SettingsModal.qml

```diff
@@ -0,0 +1,46 @@
+import QtQuick 2.3
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.3
+import Qt.labs.platform 1.1
+import "../../../imports"
+import "../../../shared"
+import "./Components"
+
+Item {
+    function open() {
+        popup.open()
+        settingsModalContent.currency = assetsModel.defaultCurrency
+    }
+
+    function close() {
+        popup.close()
+    }
+
+    Popup {
```

Comment:
Richard created a reusable component for modal popups that simplifies the process heavily. It's better to use it.

Reply:
we'll do this later together with the send modal